### DISCUSSION TITLE
Release Notes: v2.9.0

### DISCUSSION THREAD

**Discussion by PavelBlend** [PavelBlend/blender-xray#447] [General]
## Улучшения:

 - Добавлена поддержка импорта scene_objects.part файлов из X-Ray SDK 0.5, 0.6, 0.7. Формат из ТЧ SDK пока не поддерживается. Экспорт тоже пока не поддерживается:
![01](https://user-images.githubusercontent.com/7983249/146432974-887d98c9-20b5-4359-b8ef-b56acce1d340.jpg)

 - Добавлены операторы для копирования/вставки списка анимаций:
![01](https://user-images.githubusercontent.com/7983249/146431126-29d17589-aae9-4d79-8bc3-2baaa9a0a56d.jpg)

 - Добавлены операторы для копирования/вставки/удаления списка motion references:
![02](https://user-images.githubusercontent.com/7983249/146431168-1e1f9e48-011e-416c-b68a-210126af80e6.jpg)

 - Добавлен оператор для добавления motion references на основе путей к omf файлам, выбранным из файлового проводника blender (но для начала нужно указать путь к папке gamedata\meshes в настройках аддона):
![05](https://user-images.githubusercontent.com/7983249/146431262-ae50c013-7e62-48d1-8708-a8e1f019ba5b.jpg)
![04](https://user-images.githubusercontent.com/7983249/146431276-9e2ee793-004b-44f0-b466-986bcb23468c.jpg)
![03](https://user-images.githubusercontent.com/7983249/146431308-f09638ee-1c62-4f4a-b3e4-fc570719685e.jpg)

 - В настройки аддона добавлен параметр пути для gamedata\meshes папки, который используется оператором выше (для добавления motion references на основе путей):
![06](https://user-images.githubusercontent.com/7983249/146431526-e86773d5-cdc4-491e-a65a-a9e4d74c4d1f.jpg)

 - Добавлен оператор для массового (много файлов за один раз) экспорта skl файлов (в предыдущих версиях аддона была возможность экспортировать только по одному skl файлу). Экспортируются все анимации из списка Motions у объекта. Можно делать экспорт анимаций сразу из нескольких объектов. Для каждого объекта будут созданы отдельные подпапки:
![07](https://user-images.githubusercontent.com/7983249/146431703-99302d49-9174-4843-a00b-06ef03e1eda0.jpg)
![08](https://user-images.githubusercontent.com/7983249/146431726-56f5bc34-5188-4715-9e80-ec489bf79c0c.jpg)
![09](https://user-images.githubusercontent.com/7983249/146431742-a3c7692d-43d5-40cd-a5c0-aad2bebd9592.jpg)

 - Добавлены новые опции для оператора Colorize Materials, которые позволяют изменить не только цвет материала в 3d окне, но и цвет в шейдерных нодах:
![11](https://user-images.githubusercontent.com/7983249/146431971-d1070ee9-c860-46c4-8d3c-952b09906197.jpg)

 - Добавлен оператор Colorize Objects для рандомного изменения цвета объектов:
![12](https://user-images.githubusercontent.com/7983249/146432017-b3890deb-c892-47d2-bcd3-f3f9ebe8354b.jpg)

## Изменения:
 - Флаг HQ Export теперь вынесен отдельно и отображается всегда (как в SDK), а не только в режиме Custom:
![10](https://user-images.githubusercontent.com/7983249/146431909-1b6159cc-5a2f-4c79-a9cc-dfb41b22e5c5.jpg)
 - Удалены операторы для создания/удаления/скрытия фейковых костей, так как для них была создана замена в виде оператора Create Connected Bones.
 - Изменено отображение некоторых параметров в панелях, чтобы имя параметра не заслонялось полем со значением.
 - Так же внесены незначительные изменения и исправлены ошибки.

**Comment by SamArtGit** [comment]
Классно! Так и до добавления Forest Generator #343 и рабочей кнопки Load Active Motions Refs не далеко!

**Reply by PavelBlend** [reply]
> добавления Forest Generator

честно говоря, у меня нет идей, как его сделать. Я могу посмотреть скрипты Den-Stash'а, но не думаю, что что-то пойму в них.

> рабочей кнопки Load Active Motions Refs не далеко!

а сейчас она не рабочая? У меня она и так работает. Я в видео объяснял, как ею пользоваться (на 14:48):
https://youtu.be/KG5r0vRfUtk?t=888

Если у тебя всё равно не работает, то можешь поподробнее описать, как именно не работает? В настройках папка rawdata\objects указана? Ошибка в консоли есть? И версия блендера какая?
